modules/polling/api/fetchPolls.ts

```diff
@@ -162,3 +180,277 @@ export async function getPolls(
+
+  if (githubPollsHash && cachedPollsHash && githubPollsHash === cachedPollsHash) {
+    cacheSet(isPollsHashValidCacheKey, 'true', network, THIRTY_MINUTES_IN_MS, true);
+    return { valid: true };
+  } else {
+    return { valid: false, hash: githubPollsHash };
+  }
+}
+
+export async function refetchPolls(
+  network: SupportedNetworks,
+  githubHash: string
+): Promise<{
+  pollList: PollListItem[];
+  allPolls: Poll[];
+  partialActivePolls: { pollId: number; endDate: Date }[];
+}> {
+  const allPollsRes = await fetch(AGGREGATED_POLLS_FILE_URL[network]);
+  const allPolls = await allPollsRes.json();
+
+  const pollList: PollListItem[] = allPolls.map(poll => {
+    const { pollId, startDate, endDate, slug, parameters, summary, title, options, tags } = poll;
+    return {
+      pollId,
+      startDate: new Date(startDate).toISOString(),
+      endDate: new Date(endDate).toISOString(),
+      slug,
+      type: parameters.inputFormat.type,
+      parameters,
+      title,
+      summary,
+      options,
+      tags
+    };
+  });
+
+  const partialActivePolls = pollList
+    .filter(poll => new Date(poll.endDate) > new Date() && new Date(poll.startDate) <= new Date())
+    .map(({ pollId, endDate }) => ({ pollId, endDate }));
+
+  cacheSet(isPollsHashValidCacheKey, 'true', network, THIRTY_MINUTES_IN_MS, true);
+  cacheSet(pollsHashCacheKey, githubHash, network, ONE_WEEK_IN_MS, true);
+  cacheSet(pollListCacheKey, JSON.stringify(pollList), network, ONE_WEEK_IN_MS, true);
+  cacheSet(partialActivePollsCacheKey, JSON.stringify(partialActivePolls), network, ONE_WEEK_IN_MS, true);
+
+  const pollTags = getPollTags();
+
+  const pollSlugToIds: [string, number][] = [];
+  const stringifiedPolls: { [key: number]: string } = {};
+
+  const parsedPolls: Poll[] = allPolls.map((poll, i) => {
+    const prevPoll = allPolls[i - 1];
+    const nextPoll = allPolls[i + 1];
+
+    const pollWithLinksAndTags: Poll = {
+      ...poll,
+      tags: poll.tags.map(tag => pollTags.find(t => t.id === tag)).filter(tag => !!tag),
+      ctx: {
+        prev: prevPoll ? { slug: prevPoll.slug } : null,
+        next: nextPoll ? { slug: nextPoll.slug } : null
+      }
+    };
+
+    pollSlugToIds.push([poll.slug, poll.pollId]);
+    stringifiedPolls[poll.pollId] = JSON.stringify(pollWithLinksAndTags);
+
+    return pollWithLinksAndTags;
+  });
+
+  // Individual polls contain more metadata than the poll-list array and are used to render the poll detail page.
+  // They are stored as multiple hashes for a same cache key for improved performance when setting and getting them.
+  cacheSet(pollDetailsCacheKey, stringifiedPolls, network, ONE_WEEK_IN_MS, true, 'HSET');
+  cacheSet(pollSlugToIdsCacheKey, JSON.stringify(pollSlugToIds), network, ONE_WEEK_IN_MS, true);
+
+  return { pollList, allPolls: parsedPolls, partialActivePolls };
+}
+
+export async function getPartialActivePolls(network: SupportedNetworks): Promise<PartialActivePoll[]> {
```

Comment:
It's just a poll ID with the poll end date, and it's used in those parts of the app where we need to verify whether the votes in the ballot correspond to active polls. I thought it would be a better approach to fetch a list with small objects than having to fetch the full active polls list, especially thinking about a time when we could have more active polls than the max page size on the paginated polls API endpoint (30 polls max), having to make 2 API calls on the polling page.

That said, I think we may not need to use the `PartialActivePoll` on the frontend and just past the list of active poll IDs from the backend, making the API endpoint fetch the cache key with the list that may have been stored some days before, filtering out those that may have already ended and responding with an array of active poll IDs.